golang/pkg/rnr/task_nested.go

```diff
@@ -11,18 +11,29 @@ import (
 // Nested Task
 
 type NestedTask struct {
-	pbMutex     sync.Mutex
-	pb          pb.Task
-	children    []Task
-	oldState    map[*Task]pb.TaskState
-	parallelism int
+	pbMutex  sync.Mutex
+	pb       pb.Task
+	children []Task
+	oldState map[*Task]pb.TaskState
+	opts     NestedTaskOptions
 }
 
-func NewNestedTask(name string, parallelism int) *NestedTask {
+type NestedTaskOptions struct {
+	Parallelism int  // the number of tasks to run in parallel; defaults to 1
+	CompleteAll bool // if `true`, the NestedTask will attempt to run all tasks before transitioning to either SUCCEEDED or FAILED state.
+}
+
+func NewNestedTask(name string, opts *NestedTaskOptions) *NestedTask {
```

Comment:
Any reason why `opts` should be a pointer?

Reply:
Fixed :-)